### PR TITLE
SALTO-6395: fieldConfiguration in DC

### PR DESCRIPTION
**Pull request by IdoZakk**
Following a bug for a customer we limit deployment of field configurations to one at a time in Jira DC

---

None

---
_Release Notes_: 
Jira Adapter:
* Fixed a bug with FieldConfiguration deployment in jira DC

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/jira-adapter/src/filters/field_configuration/field_configuration_deployment.ts [hunk 1]
```diff
@@ -24,39 +24,51 @@ import {
...
+  })
+
 const deployFieldConfigurationItems = async (
   instance: InstanceElement,
-  client: clientUtils.HTTPWriteClientInterface,
+  client: JiraClient,
   config: JiraConfig,
 ): Promise<void> => {
-  const fields = (instance.value.fields ?? [])
+  const fields: Values[] = (instance.value.fields ?? [])
```
Comment (shayc331) [inline packages/jira-adapter/src/filters/field_configuration/field_configuration_deployment.ts:57]: nit: is this change necessary?

**Review by shayc331 — APPROVED** [review]
🤞

File: packages/jira-adapter/src/filters/field_configuration/field_configuration_deployment.ts (hunk 1 above)
Comment (IdoZakk) [reply packages/jira-adapter/src/filters/field_configuration/field_configuration_deployment.ts:57]: Yes, the build fails without it
